nodejs/eks/examples/encryption-provider/index.ts

```diff
@@ -0,0 +1,71 @@
+import * as aws from "@pulumi/aws";
+import * as eks from "@pulumi/eks";
+import * as kx from "@pulumi/kubernetesx";
+import * as pulumi from "@pulumi/pulumi";
+import * as random from "@pulumi/random";
+import * as fs from "fs";
+import * as os from "os";
+
+const projectName = pulumi.getProject();
+
+// Create a new KMS key.
+const key = new aws.kms.Key("secrets-encryption-key", {
+    deletionWindowInDays: 7,
+    description: "KMS key for encrypting Kubernetes Secrets using envelope encryption",
+});
+
+// Create a new alias to the key.
+const alias = new aws.kms.Alias("alias/k8s-secrets-encryption-key", {
+    targetKeyId: key.keyId,
+});
+
+// Export the arns
+export const keyArn = key.arn;
+export const aliasArn = alias.arn;
+
+// Create an EKS cluster with a KMS encryption provider.
+const cluster = new eks.Cluster(`${projectName}`, {
+    encryptionConfigKeyArn: alias.targetKeyArn,
+});
+
+// Export the cluster kubeconfig.
+export const kubeconfig = cluster.kubeconfig;
+
+// Create a random password for an example DB, and store it in a secret.
+const dbPassword = new random.RandomPassword("db-password",
+    {length: 32},
+    {additionalSecretOutputs: ["result"]},
+).result;
+const dbPasswordSecret = new kx.Secret("db-password", {
+    stringData: {"dbPassword": dbPassword},
+}, {provider: cluster.provider});
+
+/*
```

Comment:
Is this block still needed?

Reply:
It was left in for user reference, as image pull secrets is a common secret pattern to use this encryption provider.

It was commented out only due to CI, as it does not have a docker creds directory nor does it need one.